appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockChatBot.java

```diff
@@ -107,13 +107,19 @@ private void getTokenFromServer() {
     Ode.getInstance().getTokenAuthService().getChatBotToken(new OdeAsyncCallback<String>() {
       @Override
       public void onSuccess(String token) {
-        EditableProperty tokenProperty = MockChatBot.this.properties.getProperty(PROPERTY_NAME_TOKEN);
-        if (tokenProperty != null) {
-          String existingToken = tokenProperty.getValue();
-          if (!existingToken.isEmpty()) {
-            LOG.info("bailing on getTokenFromServer existingToken = " + existingToken);
-            return;             // If we have a value, don't over-write it
-          }
+        if (token == null) {
+          onFailure(new UnsupportedOperationException(
```

Comment:
One thing to think about is whether we want to internationalize this message or not. In theory, this error indicates that the server is misconfigured and shouldn't happen in production, but famous last words and such.

Reply:
> One thing to think about is whether we want to internationalize this message or not. In theory, this error indicates that the server is misconfigured and shouldn't happen in production, but famous last words and such.

I would say not to internationalize it. It doesn't seem like a good use of translator effort for something that is not expected to be user-facing.